### ISSUE TITLE
About Pretrained checkpoint

### ISSUE THREAD

**Issue by 6irdy** [Delong-liu-bupt/Word4Per#8]
Thank you for sharing the codes. I've tried to do inference with your provided checkpoint, but its performance is lower than the paper. Is there anything wrong with my training config? Here is the result:
{
    "model": "tuned_last_recall_at1.pt",
    "dataset": "itcpr",
    "fda_k": 6,
    "fda_alpha": 0.5,
    "fd_margin": 0.5,
    "recall_at_1": 44.77747344970703,
    "recall_at_5": 65.30426788330078,
    "recall_at_10": 72.02542877197266,
    "mAP": 54.35664749145508
}

The training config is here:
{
    "backbone": "pretrain",
    "batch_size": 256,
    "blip_model_name": "blip2_fafa_cpr",
    "exp_name": "FAFA_SynCPR_FDA_FD_MFR",
    "fd_margin": 0.5,
    "fda_alpha": 0.5,
    "fda_k": 6,
    "itcpr_root": "/data",
    "json_path": "SynCPR.json",
    "learning_rate": 2e-06,
    "loss_fd": 1.0,
    "loss_fda": 1.0,
    "loss_mfr": 0.5,
    "num_epochs": 10,
    "num_workers": 4,
    "resume_path": "null",
    "save_best": true,
    "save_last": false,
    "save_memory": false,
    "save_training": true,
    "setting": [
        "annotations"
    ],
    "syncpr_data_path": "/data/SynCPR",
    "target_ratio": 1.25,
    "transform": "squarepad",
    "validation_frequency": 1,
    "validation_step": 500
}

**Comment by Delong-liu-bupt** [comment]
Based on the configuration you've provided, everything seems to be in order. However, I believe the performance differences might be due to the following reasons:

1. **Model Selection**: Please use the `tuned_best` model instead of the `tuned_last` model for the final results. The last model is often not the optimal one, and the best-performing model is typically saved separately as `tuned_best`.

2. **Data Distribution Difference**: The training data is synthetic, while the test data is real-world data. Naturally, there will be discrepancies between the two domains. I’ve also noticed significant performance fluctuations during validation, and the random seed can have an impact (even with the same seed, results may vary across different machines). I suspect the major factor influencing performance is the difference in data distribution between the validation set and the test set. You might want to try adjusting the `validation-step=200` to see if performance improves.

3. **Data Selection**: Each folder in the test data contains the same batch of generated data. It might be helpful to try training with images from individual folders and compare their distributions. This way, you can selectively choose data that is more representative of the target distribution, potentially yielding better performance. I was in a bit of a rush during my initial training, so I didn’t focus too much on this, but spending some time on data selection should help improve results.

**Comment by 6irdy** [comment]
tuned_last_recall_at1.pt is from [download](https://drive.google.com/file/d/10sYj3bxddFhku_OKZVcvurUPQQQxXw3Q/view?usp=sharing) according to your readme in FAFA. You mean you upload the wrong version of your best ckpt?

**Comment by Delong-liu-bupt** [comment]
> tuned_last_recall_at1.pt is from [download](https://drive.google.com/file/d/10sYj3bxddFhku_OKZVcvurUPQQQxXw3Q/view?usp=sharing) according to your readme in FAFA. You mean you upload the wrong version of your best ckpt?

I misunderstood your original message and thought you were referring to training not achieving the expected performance. Regarding the inference, after checking the model I uploaded, I found that there were indeed some issues. I’ve now updated it to the correct version of the model. Thank you for pointing that out!
